### PR TITLE
`azurerm_mssql_server_transparent_data_encryption` - Support for autorotation of KV keys

### LINKED ISSUE #14171: Support for auto rotate key in azurerm_mssql_server_transparent_data_encryption
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
sql Server transparent data encryption should offer a boolean to enable key auto rotation like in the Azure UI
image

### New or Affected Resource(s)

* azurerm_mssql_server_transparent_data_encryption

### Potential Terraform Configuration

```hcl
resource "azurerm_mssql_server_transparent_data_encryption" "encrypt" {
  server_id = azurerm_mssql_server.main.id
  key_vault_key_id = key.id
  auto_rotate_key = true
}
```

### References

* #0000

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #14171

- [x] Docs updated

## Acceptance tests:
image

### CODE REVIEW

File: internal/services/mssql/mssql_server_transparent_data_encryption_resource.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ func resourceMsSqlTransparentDataEncryption() *pluginsdk.Resource {
 				Optional:     true,
 				ValidateFunc: keyVaultValidate.NestedItemId,
 			},
+			"auto_rotation_enabled": {
```
Comment (aristosvo) [inline internal/services/mssql/mssql_server_transparent_data_encryption_resource.go:53]:
```suggestion
			},

			"auto_rotation_enabled": {
```

File: internal/services/mssql/mssql_server_transparent_data_encryption_resource.go [hunk 2]
```diff
@@ -45,11 +45,18 @@ func resourceMsSqlTransparentDataEncryption() *pluginsdk.Resource {
...
 			"key_vault_key_id": {
 				Type:         pluginsdk.TypeString,
 				Optional:     true,
 				ValidateFunc: keyVaultValidate.NestedItemId,
 			},
+
+			"auto_rotation_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Computed: true,
```
Comment (tombuildsstuff) [inline internal/services/mssql/mssql_server_transparent_data_encryption_resource.go:58]: out of interest why make this computed? this'll have a default value of `false`, which'll be returned via `d.Get` below so we may as well use that as the default here and set it via a state migration?

**Review by tombuildsstuff — APPROVED** [review]
One question from me around the default value (since we're getting the boolean default value `false` via `d.Get` and defaulting this to `false` in the Read function) - but this otherwise LGTM 👍

File: internal/services/mssql/mssql_server_transparent_data_encryption_resource.go (hunk 2 above)
Comment (aristosvo) [reply internal/services/mssql/mssql_server_transparent_data_encryption_resource.go:58]: Sure! I've never done a state migration, but I can definitely a try!

**Pull request merged**